### PR TITLE
Add medias with title, url and thumbnails to index

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| License | MIT
| Documentation PR | sulu/sulu-docs#prnum

#### What's in this PR?

Add medias with title, url and thumbnails to index

#### Why?

So it need only to load the media when a article is changed.

### CODE REVIEW

File: Document/MediaCollectionOngrObject.php [hunk 1]
```diff
@@ -11,15 +11,18 @@
...
+use ONGR\ElasticsearchBundle\Collection\Collection;
+use Sulu\Bundle\MediaBundle\Api\Media;
 
 /**
  * Contains the ids and display-options.
  *
  * @Object
  */
-class MediaCollectionOngrObject
+class MediaCollectionOngrObject implements \Iterator, \ArrayAccess
```
Comment (wachterjohannes) [inline Document/MediaCollectionOngrObject.php]: use this (http://php.net/manual/de/class.iteratoriterator.php) as parent class which implements all the necessary functionality for you.

File: Document/MediaOngrObject.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    /**
+     * @var string
+     *
+     * @Property(type="string", options={"index"="not_analyzed"})
+     */
+    public $url;
+
+    /**
+     * @param Media $media
```
Comment (wachterjohannes) [inline Document/MediaOngrObject.php]: Use this phpdoc:

```
Set data.

@param Media $media

@return $this
```

File: Document/MediaOngrObject.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    {
+        $this->id = $media->getId();
+        $this->title = $media->getTitle();
+        $this->setFormats($media->getFormats());
+        $this->url = $media->getUrl();
+        $this->copyright = $media->getCopyright();
+    }
+
+    /**
+     * @return string
```
Comment (wachterjohannes) [inline Document/MediaOngrObject.php]: use this phpdoc:

```
Returns formats.

@return string[]
```

and update phpdoc for all the new getter and setter.

**Comment by wachterjohannes** [comment]
@alexander-schranz please rebase this PR and fix the tests (:

File: Document/MediaCollectionOngrObject.php (hunk 1 above)
Comment (alexander-schranz) [inline Document/MediaCollectionOngrObject.php]: needs to set it inside the construct and this will not work with ONGR.

**Comment by alexander-schranz** [comment]
@wachterjohannes fixed

**Pull request merged**